src/neural/loader.cc

```diff
@@ -108,6 +108,20 @@ void FixOlderWeightsFile(WeightsFile* file) {
   using nf = pblczero::NetworkFormat;
   auto network_format = file->format().network_format().network();
   const auto has_network_format = file->format().has_network_format();
+  // Hack for old encoding compatibility. REMOVE BEFORE RELEASE.
```

Comment:
What is exactly the reason of this hack, in particular,

- What is the harm of keeping it after the release?
- Why is it fine to have in master before the release?

If there's indeed a reason to keep it in master before the release but remove after, maybe add a version check that throws the exception (or even better, some kind of static assert), so that we don't forget about it.

Reply:
This was also merged with #1858, it is for compatibility with experimental attention body/smolgen nets that were trained before the net.proto update, in particular many BT2 nets and some distilled ones that people still use. As newer nets come out there should be less need for this so it can be removed, but for now it is a common issue. There is no harm keeping it indefinitely, but as those are unofficial nets it didn't seem worthwhile.